src/Stmt.h

```diff
@@ -516,9 +516,12 @@ class AssertStmt final : public Stmt {
     StmtPtr DoReduce(Reducer* c) override;
 
 private:
-    ExprPtr cond;
     std::string cond_desc;
     ExprPtr msg;
+
+    // Statement to execute before evaluating "msg". Only used for script
+    // optimization.
+    StmtPtr msg_setup_stmt;
```

Comment:
I'm not following if this is left-over from attempted exception handling, or required because it'll be set during `DoReduce()` via:
```
        msg = msg->ReduceToSingleton(c, msg_setup_stmt);
```

I'm just making things up, but could `AssertStmt::DoReduce()` return a `StmtList` prepended with the setup stmt?

This isn't a biggie, it's just that it's one more optimization specific field/accessor and I find it not easy to follow why it exists - if you say no dice or complicated, no worries.

Reply:
Yeah, it needs to be separate. The reason is that it should only be evaluated if the assertion is going to be reported, and the statement might not be valid if it's not going to be reported. For example,
```
assert ! r?$b, fmt("r$b unexpectedly set: %s", r$b);
```

Reply:
Hmmm I may see a bug in how that's implemented. However I'd rather tackle that post-merge, if you're okay with that.